File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle) {
    int vectorsize = triangle.size() - 1; // 0~499;
    while(vectorsize > 0){
    for(int i = 0; i < vectorsize ; i++){
            triangle[vectorsize-1][i] += max(triangle[vectorsize][i], triangle[vectorsize][i + 1]);
        }
        vectorsize--;
    }
    return triangle[0][0];
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.Arrays;
class Solution {
    public int solution(int[][] triangle) {
    int t_size = triangle.length - 1;
        while(t_size > 0){
            for(int i = 0; i < t_size; i++){
                triangle[t_size-1][i] += Math.max(triangle[t_size][i], triangle[t_size][i + 1]); 
            }
            t_size--;
        }
    return triangle[0][0];
    }
}